### PR TITLE
Added check for volumes in vmgroup on removal.

### PR DESCRIPTION
**Pull request by govint**
Added the check on the presence of volumes in a vmgroup and disallow the deletion till user moves the datastores off the tenant or provides the remove-volumes option to the user.

python /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py vmgroup ls
Uuid                                  Name      Description                Default_datastore  VM_list
------------------------------------  --------  -------------------------  -----------------  -------
11111111-1111-1111-1111-111111111111  _DEFAULT  This is a default vmgroup  _VM_DS
55e5f60f-f536-4571-9f82-3d8acddc06ab  vmg1                                 sharedVmfs-1
f961d62f-0801-42b5-96c0-beb60d4323bb  vmg2                                 sharedVmfs-1
80a49fcd-ad12-4d32-a784-9835298f4982  vmg3                                 sharedVmfs-1
a47118cd-9359-4986-8ce5-930fbde32cf1  vmg4                                 sharedVmfs-1
82adf657-fdce-4485-80c7-bb4104fb6db3  vmg5                                 sharedVmfs-1

python /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py volume ls
Volume     Datastore     VMGroup   Capacity  Used  Filesystem  Policy  Disk Format  Attached-to    Access      Attach-as               Created By     Created Date
---------  ------------  --------  --------  ----  ----------  ------  -----------  -------------  ----------  ----------------------  -------------  ------------------------
_vF_fvol2  sharedVmfs-0  _DEFAULT  100MB     17MB  ext4        N/A     thin         worker2-VM2.0  read-write  independent_persistent  worker1-VM1.0  Thu Sep  7 10:12:29 2017
abc        sharedVmfs-0  _DEFAULT  100MB     15MB  ext4        N/A     thin         worker2-VM2.0  read-write  independent_persistent  worker2-VM2.0  Tue Sep 12 12:39:15 2017
xyz        sharedVmfs-0  _DEFAULT  100MB     13MB  ext4        N/A     thin         detached       read-write  independent_persistent  master-VM0.0   Tue Sep 12 13:46:58 2017
notthere   sharedVmfs-0  _DEFAULT  100MB     15MB  ext4        N/A     thin         detached       read-write  independent_persistent  master-VM0.0   Wed Sep 13 05:30:15 2017
dsname     sharedVmfs-0  _DEFAULT  100MB     13MB  ext4        N/A     thin         detached       read-write  independent_persistent  master-VM0.0   Fri Oct  6 10:04:59 2017
dsname2    sharedVmfs-0  _DEFAULT  100MB     13MB  ext4        N/A     thin         detached       read-write  independent_persistent  master-VM0.0   Mon Oct  9 04:45:06 2017

python /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py vmgroup rm --name _DEFAULT
ERROR:Internal Error(The vmgroup has volumes in it, these must be removed/migrated before deleting the vmgroup.)

### CODE REVIEW

**Comment by shuklanirdesh82** [comment]
@govint There is a CI failure with this PR, can you please run test locally first and update the PR description with `Testing Done` section?

```
----------------------------------------------------------------------
FAIL: defaultvmgroup_test.go:117: DefaultVMGroupTestSuite.TestDeleteDefaultVmgroup

defaultvmgroup_test.go:142:
    c.Assert(isVmgroupAvailable, Equals, false, Commentf("Failed to delete vmgroup %s .", admincliconst.DefaultVMgroup))
... obtained bool = true
... expected bool = false
... Failed to delete vmgroup _DEFAULT .
```

**Comment by govint** [comment]
CI passed on Linux and windows plugin deploy failed, does that count as CI failure still?

=> Installing plugin as a windows service on root@192.168.31.71... Wed Oct 11 09:19:13 UTC 2017


make[1]: *** [deploy-windows-plugin] Error 255
make: *** [deploy-windows-plugin] Error 2

File: tests/e2e/vmgroups_test.go [hunk 1]
```diff
@@ -674,7 +674,7 @@ func (vg *VmGroupTest) TestVmgroupRemoveWithRemoveVol(c *C) {
 // 	}
 
 // 	log.Printf("Removing test vmgroup %s", vgTestVMgroup1)
-// 	out, err = adminutils.DeleteVMgroup(vg.config.EsxHost, vgTestVMgroup1, false)
+// 	out, err = adminutils.DeleteVMgroup(vg.config.EsxHost, vgTestVMgroup1, true)
```
Comment (shuklanirdesh82) [inline tests/e2e/vmgroups_test.go]: shouldn't this test be removed? with reference `TODO: Need to enable or remove the following tests after we have conclusion on issue #1469` @line#632

**Comment by shuklanirdesh82** [comment]
Adding @shaominchen to provide 2nd review.

File: tests/e2e/vmgroups_test.go (hunk 1 above)
Comment (shaominchen) [reply tests/e2e/vmgroups_test.go]: +1

**Pull request merged**